### PR TITLE
fix: Zarr part mda

### PR DESCRIPTION
**Pull request by wl-stepp**
The zarr writer fails for me if I have only a partly set up MDASequence. Does it fail for you as well?

Will see if I can fix it.

### CODE REVIEW

**Comment by wl-stepp** [comment]
The tests pass for me with this locally.

File: tests/test_handlers/test_ome_zarr.py [hunk 1]
```diff
@@ -15,17 +15,28 @@
...
+    stage_positions=[(222, 1, 1), (111, 0, 0)],
+    z_plan={"range": 0.3, "step": 0.1},
+    axis_order="tpcz",
+)
+
+part_mda = useq.MDASequence(
+    channels=["Cy5", "FITC"],
+    stage_positions=[(222, 1, 1), (111, 0, 0)],
+    time_plan={"interval": 0.1, "loops": 3},
+)
```
Comment (tlambert03) [inline tests/test_handlers/test_ome_zarr.py:30]:
```suggestion
part_mda = useq.MDASequence(
     channels=["Cy5", "FITC"],
     stage_positions=[(222, 1, 1), (111, 0, 0)],
     time_plan={"interval": 0.1, "loops": 3},
 )
 
 full_mda = part_mda.replace(axis_order="tpcz", channels=["Cy5", "FITC"])
```

just a nice syntax to extend or build off of another MDA

File: src/pymmcore_plus/mda/handlers/_ome_zarr_writer.py [hunk 2]
```diff
@@ -228,8 +228,11 @@ def frameReady(
 
             # create the new array, getting XY chunksize from the frame
             # and total shape from the sequence
-            shape = (*tuple(v for k, v in seq.sizes.items() if k != "p"), *frame.shape)
-            axes = (*(k for k in seq.sizes if k != "p"), "y", "x")
+            shape = (
+                *tuple(v for k, v in seq.sizes.items() if k != "p" and v > 0),
+                *frame.shape,
+            )
+            axes = (*(k for k, v in seq.sizes.items() if k != "p" and v > 0), "y", "x")
```
Comment (tlambert03) [inline src/pymmcore_plus/mda/handlers/_ome_zarr_writer.py:235]:
```suggestion
            axes = (*(k for k in seq.used_axes if k != "p"), "y", "x")
            shape = (*tuple(v for k, v in seq.sizes.items() if k in axes), *frame.shape)
```

how about this?  There's an attribute called `used_axes` that takes core of these "empty" axes.

File: tests/test_handlers/test_ome_zarr.py [hunk 3]
```diff
@@ -15,17 +15,22 @@
...
-        axis_order="tpcz",
-    )
-
+part_mda = useq.MDASequence(
+     channels=["Cy5", "FITC"],
+     stage_positions=[(222, 1, 1), (111, 0, 0)],
+     time_plan={"interval": 0.1, "loops": 3},
+ )
+
+ full_mda = part_mda.replace(axis_order="tpcz", channels=["Cy5", "FITC"])
```
Comment (tlambert03) [inline tests/test_handlers/test_ome_zarr.py:24]:
```suggestion
full_mda = part_mda.replace(axis_order="tpcz", channels=["Cy5", "FITC"])
```

whoops... my bad

**Comment by tlambert03** [comment]
btw, love the test-driven development with the first commit here [e1b5fe7](https://github.com/pymmcore-plus/pymmcore-plus/pull/302/commits/e1b5fe79a18d14eae299ac93a0f643f29946b036) :)

**Pull request merged**